src/storage/src/render/upsert.rs

```diff
@@ -487,6 +488,20 @@ where
                 }
             }
 
+            for (_, value, diff) in events.iter_mut() {
+                if let Err(UpsertError::Value(ref mut err)) = value {
+                    // If we receive a legacy error in the snapshot we will keep a note of it but
+                    // insert a non-legacy error in our state. This is so that if this error is
+                    // ever retracted we will correctly retract the non-legacy version because by
+                    // that time we will have emitted the error correction, which happens before
+                    // processing any of the new source input.
+                    if err.is_legacy_dont_touch_it {
+                        legacy_errors_to_correct.push((err.clone(), diff.clone()));
```

Comment:
technically this is an unbounded buffer, so a pathological spill2disk instance could oom...probably not worth working around

Also, is the idea that retracted errors will have 2 opposite retractions sent below?

Reply:
I need to add a consolidation step below but the idea is that with whatever diff a legacy errors comes in with we must emit it with that diff negated. So yes, if we see `(legacy_error, -1)` we will emit `(legacy_error, +1, new_error, -1)`. One way this can happen is if we successfully submit the corrections, then we crash, then on restart we read a snapshot that includes the original legacy error AND our legacy error correction. So we must not assume anything about diffs